src/pages/studyView/resources/FilesAndLinks.tsx

```diff
@@ -0,0 +1,332 @@
+                    resourcesPerPatient: 0,
+                    typeOfResource: resource?.resourceDefinition?.displayName,
+                    description: resource?.resourceDefinition?.description,
+                    url: resource?.url,
+                });
+            });
+
+            if (data && data.length > 0) {
+                if (!(data[0].patientId in resourcesPerPatient))
+                    resourcesPerPatient[data[0].patientId] = 0;
+
+                resourcesPerPatient[data[0].patientId] += data.length;
+            }
+        });
+    };
+
+    itemBuilder(sampleIdResourceData);
+    const resourcesForPatients = await fetchResourceDataOfPatient(patientIds);
+    itemBuilder(resourcesForPatients);
+
+    const data = _.mapValues(items, item => {
+        item.resourcesPerPatient = resourcesPerPatient[item.patientId];
+        return item;
+    });
+
+    const sortedData = _.orderBy(data, 'resourcesPerPatient', 'desc');
+    return {
+        totalItems: sortedData.length,
+        data: _.values(sortedData),
+    };
+}
+
+@observer
+export class FilesAndLinks extends React.Component<IFilesLinksTable, {}> {
+    constructor(props: IFilesLinksTable) {
+        super(props);
+        makeObservable(this);
+    }
+
+    getDefaultColumnConfig(
+        key: string,
+        columnName: string,
+        isNumber?: boolean
+    ) {
+        return {
+            name: columnName || '',
+            headerRender: (data: string) => (
+                <span data-test={data}>{data}</span>
+            ),
+            render: (data: { [id: string]: string }) => {
+                if (isUrl(data[key])) {
+                    return (
+                        <a href={data[key]} target="_blank">
+                            {data[key]}
+                        </a>
+                    );
+                }
+                return <span data-test={data[key]}>{data[key]}</span>;
+            },
+            download: (data: { [id: string]: string }) => data[key] || '',
+            sortBy: (data: { [id: string]: any }) => {
+                if (data[key]) {
+                    if (isNumber) {
+                        return parseFloat(data[key]);
+                    } else {
+                        return data[key];
+                    }
+                }
+                return null;
+            },
+            filter: (
+                data: { [id: string]: string },
+                filterString: string,
+                filterStringUpper: string
+            ) => {
+                if (data[key]) {
+                    if (!isNumber) {
+                        return (data[key] || '')
+                            .toUpperCase()
+                            .includes(filterStringUpper);
+                    }
+                }
+
+                return false;
+            },
+        };
+    }
+
+    @observable searchTerm: string | undefined = undefined;
+
+    readonly getData = remoteData({
```

Comment:
`getData` is a confusing name because this is not a function. It's better to name it as `sampleData` or something similar.
```suggestion
    readonly sampleData = remoteData({
```

Reply:
Done